pynamodb/attributes.py

```diff
@@ -820,18 +820,15 @@ def deserialize(self, values):
         Decode as a dict.
         """
         deserialized_dict: Dict[str, Any] = dict()
-        for k in values:
-            v = values[k]
-            attr_value = _get_value_for_deserialize(v)
-            key = self._dynamo_to_python_attr(k)
-            attr_class = self._get_deserialize_class(key, v)
-            if key is None or attr_class is None:
-                continue
-            deserialized_value = None
-            if attr_value is not None:
-                deserialized_value = attr_class.deserialize(attr_value)
-
-            deserialized_dict[key] = deserialized_value
+        for k, v in values.items():
+            attr_name = self._dynamo_to_python_attr(k)
+            attr_type, attr_value = next(iter(v.items()))
+            attr_class = self._get_deserialize_class(attr_name, attr_type)
+            if attr_class and attr_class.attr_type != attr_type:
+                raise ValueError("Cannot deserialize '{}' attribute from type: {}".format(
+                    attr_name, attr_type))
+            if attr_class:
```

Comment:
nit - nest the ifs?

Reply:
replaced by splitting deserialization into raw and non-raw variants